Mathlib/RingTheory/Polynomial/DegreeLT.lean

```diff
@@ -0,0 +1,191 @@
+/-
+Copyright (c) 2025 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anne Baanen, Kenny Lau
+-/
+
+import Mathlib.Algebra.Polynomial.Div
+import Mathlib.Algebra.Polynomial.Taylor
+import Mathlib.LinearAlgebra.Determinant
+import Mathlib.LinearAlgebra.Matrix.Block
+import Mathlib.RingTheory.Polynomial.Basic
+
+/-!
+# Polynomials with degree less than `n`
+
+This file contains the properties of the submodule of polynomials of degree less than `n` in a
+(semi)ring `R`, denoted `R[X]_n`.
+
+## Main definitions/lemmas
+
+* `degreeLT.basis R n`: a basis for `R[X]_n` the submodule of polynomials with degree `< n`,
+  given by the monomials `X^i` for `i < n`.
+
+* `degreeLT.basisProd R m n`: a basis for `R[X]_m × R[X]_n`, which is the sum of two instances of
+  the basis given above.
+
+* `degreeLT.addLinearEquiv R m n`: an isomorphism between `R[X]_(m+n)` and `R[X]_m × R[X]_n`,
+  given by the fact that the bases are both indexed by `Fin (m+n)`. This is used for the Sylvester
+  matrix, which is the matrix representing the Sylvester map between these two spaces, in a future
+  file.
+
+* `taylorLinear r n`: The linear automorphism induced by `taylor r` on `R[X]_n` which sends `X` to
+  `X + r` and preserves degrees.
+
+-/
+
+namespace Polynomial
+
+@[inherit_doc] scoped notation:9000 R "[X]_" n:arg => Polynomial.degreeLT R n
```

Comment:
Can you move this notation next to the definition of `degreeLT`? And are you sure this should be notation for `degreeLT` and not `degreeLE`?

Reply:
degreeLT n has dimension (i.e. rank) n